netlogo-gui/src/main/app/WidgetActions.scala

```diff
@@ -0,0 +1,166 @@
+package org.nlogo.app
+
+import java.awt.Rectangle
+import javax.swing.JComponent
+import javax.swing.undo.{AbstractUndoableEdit, UndoManager, UndoableEdit}
+
+import org.nlogo.core.Widget
+import org.nlogo.swing.ToolBarButton
+import org.nlogo.window.MouseMode
+
+object WidgetActions {
+
+  private var counter = 0
+  private val undoManager = new UndoManager() {
+    override def addEdit(anEdit: UndoableEdit): Boolean = {
+      undoButton.setEnabled(true)
+      super.addEdit(anEdit)
+    }
+  }
+
+  var undoButton : ToolBarButton = null
```

Comment:
I think you can get around needing to operate on `undoButton` / `redoButton` just by operating on the `undo`/`redo` actions. You can set an icon for an action and use `setEnabled(isEnabled)` to enable / disable an action (which should enable/disable the button).